src/main/kotlin/graphics/scenery/volumes/Volume.kt

```diff
@@ -760,22 +766,38 @@ open class Volume(
 
             val start = System.nanoTime()
 
-//            if(reader.openPlane(0, 0).imageMetadata.isLittleEndian) {
             logger.debug("Volume is little endian")
-            (0 until reader.getPlaneCount(0)).forEach { plane ->
-                imageData.put(reader.openPlane(0, plane).bytes)
-            }
-//            } else {
-//                logger.info("Volume is big endian")
-//                (0 until reader.getPlaneCount(0)).forEach { plane ->
-//                    imageData.put(swapEndianUnsafe(reader.openPlane(0, plane).bytes))
-//                }
-//            }
+            val planeSize = bytesPerVoxel * dims.x * dims.y
+            if(hub.get<Settings>()?.get("Volume.ParallelReads", true) == true) {
+                val readers = ConcurrentHashMap<Thread, ReaderFilter>()
+                (0 until reader.getPlaneCount(0)).forEachIndexedAsync { index, plane ->
+                    val thread = Thread.currentThread()
+                    val myReader =
+                        readers.getOrPut(thread) { scifio.initializer().initializeReader(FileLocation(file.toFile())) }
+                    logger.info("${readers.size} readers available")
+
+                    val bytes = myReader.openPlane(0, plane).bytes
+                    val view = imageData.duplicate().order(ByteOrder.LITTLE_ENDIAN)
+
+                    view.position(index * planeSize)
+                    view.put(bytes)
+                }
 
-            val duration = (System.nanoTime() - start) / 10e5
-            logger.debug("Reading took $duration ms")
+                val duration = (System.nanoTime() - start) / 10e5
+                logger.debug("Reading took $duration ms, used $readers parallel readers.")
+                readers.forEach { it.value.close() }
+                readers.clear()
+            } else {
+                (0 until reader.getPlaneCount(0)).forEach { plane ->
+                    val bytes = reader.openPlane(0, plane).bytes
+                    val view = imageData.duplicate().order(ByteOrder.LITTLE_ENDIAN)
+                    view.put(bytes)
+                }
+
+                val duration = (System.nanoTime() - start) / 10e5
+                logger.debug("Reading took $duration ms, no parallel readers.")
+            }
 
-            imageData.flip()
```

Comment:
where is the pointer without this line? What is the convention where to leave a pointer after one is done with a buffer?

Reply:
We're only operating on a duplicate, so imageData's position etc. remains at the original values.